src/main/java/ar/com/carloscurotto/storm/complex/topology/propagator/gloss/MessageSender.java

```diff
@@ -0,0 +1,118 @@
+package ar.com.carloscurotto.storm.complex.topology.propagator.gloss;
+
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.transform.stream.StreamResult;
+
+import org.apache.commons.lang3.Validate;
+
+import ar.com.carloscurotto.storm.complex.service.OpenAwareService;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.ExcpTradeStatusMessage;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.NormalTradeStatusMessage;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.TradeCommentsMessage;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.TradeMessage;
+import ar.com.carloscurotto.storm.complex.transport.Producer;
+
+/**
+ * Sends the message into the queue. Knows what queue to use and knows how to marshall the messages.
+ *
+ * @author D540601
+ */
+public class MessageSender extends OpenAwareService<List<TradeMessage>, Void> {
+
+    protected Map<Class<? extends TradeMessage>, Marshaller> marshallers = new HashMap<Class<? extends TradeMessage>, Marshaller>();
+
+    private Producer<String> messageProducer;
+
+    /**
+     * Constructs the MessageSender with the give message producer.
+     * 
+     * @param theMessageProducer
+     *            a {@link Producer<String>} that will be used to send the update messages to the consumer of the
+     *            producer.
+     */
+    public MessageSender(final Producer<String> theMessageProducer) {
+        Validate.notNull(theMessageProducer, "The message producer cannot be null");
+        messageProducer = theMessageProducer;
+    }
+
+    /**
+     * Creates the marshallers for the trade status messages classes and stores them in the marshallers map.
+     */
+    @Override
+    protected void doOpen() {
+        try {
+            marshallers.put(NormalTradeStatusMessage.class,
```

Comment:
Analicé si era conveniente usar los contextos JAXB asi como los marshallers como spring beans pero fuera de del codigo del open no tienen relevancia asi que es mejor pasarle al constructor las clases que quiero que sean capaces de ser aplanadas (de alguna manera la voz pasiva de "to marshall" traducido al español).
Tambien analice si era conveniente poner una anotacion en las clases pertinentes y por reflection analizar las clases que tuvieran tal anotacion y usar esas clases, pero es lo mismo (aplicando un criterio) que tener siempre todo por codigo. Si por otro lado hubiera elegido usar spring para pasar el package de las clases anotadas a analizar, seria de alguna manera parecido a pasar la lista de clases, solo que menos flexible y mucho mas complejo de implementar.
fixed.